exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillSqlParser.java

```diff
@@ -0,0 +1,349 @@
+  }
+
+  public SchemaPlus getDefaultSchema() {
+    return defaultSchema;
+  }
+
+  private class DrillValidator extends SqlValidatorImpl {
+    protected DrillValidator(SqlOperatorTable opTab, SqlValidatorCatalogReader catalogReader,
+        RelDataTypeFactory typeFactory, SqlConformance conformance) {
+      super(opTab, catalogReader, typeFactory, conformance);
+    }
+  }
+
+  private static class DrillTypeSystem extends RelDataTypeSystemImpl {
+
+    @Override
+    public int getDefaultPrecision(SqlTypeName typeName) {
+      switch (typeName) {
+      case CHAR:
+      case BINARY:
+      case VARCHAR:
+      case VARBINARY:
+        return 65536;
+      default:
+        return super.getDefaultPrecision(typeName);
+      }
+    }
+
+    @Override
+    public int getMaxNumericScale() {
+      return 38;
+    }
+
+    @Override
+    public int getMaxNumericPrecision() {
+      return 38;
+    }
+
+  }
+
+  public RelNode toRel(
+      final SqlNode validatedNode) {
+    final RexBuilder rexBuilder = new RexBuilder(typeFactory);
+    if (planner == null) {
+      planner = new VolcanoPlanner(costFactory, settings);
+      planner.setExecutor(new DrillConstExecutor(functions, util, settings));
+      planner.clearRelTraitDefs();
+      planner.addRelTraitDef(ConventionTraitDef.INSTANCE);
+      planner.addRelTraitDef(DrillDistributionTraitDef.INSTANCE);
+      planner.addRelTraitDef(RelCollationTraitDef.INSTANCE);
+    }
+
+    final RelOptCluster cluster = RelOptCluster.create(planner, rexBuilder);
+    final SqlToRelConverter sqlToRelConverter =
+        new SqlToRelConverter(new Expander(), validator, catalog, cluster, DrillConvertletTable.INSTANCE);
+
+    sqlToRelConverter.setTrimUnusedFields(false);
+    sqlToRelConverter.enableTableAccessConversion(false);
+    final RelNode rel = sqlToRelConverter.convertQuery(validatedNode, false, true);
+    final RelNode rel2 = sqlToRelConverter.flattenTypes(rel, true);
+    final RelNode rel3 = RelDecorrelator.decorrelateQuery(rel2);
+    return rel3;
+
+  }
+
+  private class Expander implements RelOptTable.ViewExpander {
+
+    public Expander() {
+    }
+
+    public RelNode expandView(
+        RelDataType rowType,
+        String queryString,
+        List<String> schemaPath) {
+      DrillSqlParser parser = new DrillSqlParser(DrillSqlParser.this, defaultSchema, rootSchema,
+          catalog.withSchemaPath(schemaPath));
+      return expandView(queryString, parser);
+    }
+
+    @Override
+    public RelNode expandView(
+        RelDataType rowType,
+        String queryString,
+        SchemaPlus rootSchema, // new root schema
+        List<String> schemaPath) {
+      final CalciteCatalogReader catalogReader = new CalciteCatalogReader(
+          CalciteSchemaImpl.from(rootSchema),
+          parserConfig.caseSensitive(),
+          schemaPath,
+          typeFactory);
+      SchemaPlus schema = rootSchema;
+      for (String s : schemaPath) {
+        schema = schema.getSubSchema(s);
```

Comment:
Is it possible that schema.getSubSchema(s) could return null? In such case, the next loop iteration will possibly hit NPE.